conventionalcommits/type.go

```diff
@@ -16,30 +16,93 @@ const (
 )
 
 var (
-	choreTypes   = []string{"build", "chore", "ci", "docs", "style", "refactor", "perf", "test"}
-	fixTypes     = []string{"fix"}
-	featureTypes = []string{"feat"}
-	allTypes     []string
+	defaultChoreTypes   = []string{"build", "chore", "ci", "docs", "style", "refactor", "perf", "test"}
+	defaultFixTypes     = []string{"fix"}
+	defaultFeatureTypes = []string{"feat"}
 )
 
-func initType() {
-	for _, types := range [][]string{choreTypes, fixTypes, featureTypes} {
-		allTypes = append(allTypes, types...)
+type TypeClassifier struct {
+	choreTypes   []string
+	fixTypes     []string
+	featureTypes []string
+	allTypes     []string
+}
+
+func NewTypeClassifier() *TypeClassifier {
+	tc := &TypeClassifier{
+		choreTypes:   append([]string{}, defaultChoreTypes...),
+		fixTypes:     append([]string{}, defaultFixTypes...),
+		featureTypes: append([]string{}, defaultFeatureTypes...),
 	}
+	tc.updateAllTypes()
+	return tc
 }
 
-func StringToType(s string) (Type, error) {
-	if slices.Contains(choreTypes, strings.ToLower(s)) {
-		return Chore, nil
+func NewTypeClassifierWithCustomPrefixes(customChoreTypes, customFixTypes, customFeatureTypes []string) *TypeClassifier {
+	tc := &TypeClassifier{}
+	
+	if len(customChoreTypes) > 0 {
+		tc.choreTypes = customChoreTypes
+	} else {
+		tc.choreTypes = append([]string{}, defaultChoreTypes...)
+	}
+	
+	if len(customFixTypes) > 0 {
+		tc.fixTypes = customFixTypes
+	} else {
+		tc.fixTypes = append([]string{}, defaultFixTypes...)
 	}
+	
+	if len(customFeatureTypes) > 0 {
+		tc.featureTypes = customFeatureTypes
+	} else {
+		tc.featureTypes = append([]string{}, defaultFeatureTypes...)
+	}
+	
+	tc.updateAllTypes()
+	return tc
+}
 
-	if slices.Contains(fixTypes, strings.ToLower(s)) {
-		return Fix, nil
+func (tc *TypeClassifier) updateAllTypes() {
+	tc.allTypes = nil
+	for _, types := range [][]string{tc.choreTypes, tc.fixTypes, tc.featureTypes} {
+		tc.allTypes = append(tc.allTypes, types...)
 	}
+}
 
-	if slices.Contains(featureTypes, strings.ToLower(s)) {
+func (tc *TypeClassifier) GetAllTypes() []string {
+	return tc.allTypes
+}
+
+func (tc *TypeClassifier) StringToType(s string) (Type, error) {
+	lowerS := strings.ToLower(s)
+	
+	if slices.Contains(tc.featureTypes, lowerS) {
 		return Feature, nil
 	}
 
+	if slices.Contains(tc.fixTypes, lowerS) {
+		return Fix, nil
+	}
+
+	if slices.Contains(tc.choreTypes, lowerS) {
+		return Chore, nil
+	}
+
 	return Chore, errors.New("invalid string for conventional commit type")
 }
+
+var defaultClassifier = NewTypeClassifier()
```

Comment:
I would like to get rid of the `defaultClassifier`, so that from the outside you just create a type classifier, where you always hand over the custom prefixes – but they can be empty.